packages/core/src/html/UlNodeProcessor.ts

```diff
@@ -0,0 +1,256 @@
+import cheerio from 'cheerio';
+import { MbNode, NodeOrText } from '../utils/node';
+
+const { processIconString } = require('../lib/markdown-it/plugins/markdown-it-icons');
+const emojiDictionary = require('../lib/markdown-it/patches/markdown-it-emoji-fixed');
+
+interface EmojiData {
+  [key: string]: string;
+}
+
+const emojiData = emojiDictionary as unknown as EmojiData;
+
+interface IconAttributes {
+  icon?: string;
+  className?: string;
+  size?: string;
+  width?: string;
+  height?: string;
+}
+
+function classifyIcon(icon: string) {
+  const isEmoji = Object.prototype.hasOwnProperty.call(emojiData, icon);
+  const localFileRegex = /^(\.\/)?[a-zA-Z0-9\-._~:/?#[\]@!$&'()*+,;=%]+\.(jpg|png|gif|bmp|svg|jpeg)$/;
+  const urlRegex = /^(http(s)?|ftp):\/\/[^\s/$.?#].[^\s]*$/;
+  const isImage = !isEmoji && (localFileRegex.test(icon) || urlRegex.test(icon));
+  return {
+    isEmoji,
+    isImage,
+    unicodeEmoji: isEmoji
+      ? emojiData[icon]
+      : undefined,
+  };
+}
+
+function createIconSpan(iconAttrs: IconAttributes): cheerio.Cheerio {
+  const {
+    isEmoji,
+    isImage,
+    unicodeEmoji,
+  } = classifyIcon(iconAttrs.icon!);
+
+  let spanContent;
+
+  if (isEmoji) {
+    spanContent = `<span aria-hidden="true">${unicodeEmoji}</span>`;
+  } else if (isImage) {
+    const img = cheerio(`<img src='${iconAttrs.icon}' alt='Icon'>`)
+      .css({ width: iconAttrs.width, height: iconAttrs.height, display: 'inline-block' })
+      .addClass(iconAttrs.className || '');
+    img.append('\u200B');
+
+    spanContent = cheerio('<span></span>').append(img).css({
+      'padding-bottom': '0.3em',
+      'padding-top': '0.3em',
+    });
+  } else {
+    spanContent = processIconString(iconAttrs.icon);
+  }
+
+  let spanNode = cheerio(spanContent).css({ 'font-size': 'unset', 'min-width': '16px' });
+
+  if (!isImage) {
+    spanNode = spanNode.css({ 'font-size': iconAttrs.size }).addClass(iconAttrs.className || '');
+  }
+  // Add invisible character to avoid the element from being empty
+  spanNode.append('\u200B');
+  return spanNode.css({
+    'line-height': 'unset',
+    'margin-inline-end': '0.3em',
+    height: '100%',
```

Comment:
This ensures that the icon span takes up 100% of the icon height instead of the parent div container.

<img width="280" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/ffa940af-64e4-4129-b2b1-bbefc4d77790">

With height = `100%`
<img width="245" alt="image" src="https://github.com/MarkBind/markbind/assets/30100720/c8cf8ff3-4a14-492f-a367-f99eb69c8631">

The `height` property will not affect the font size of images as they are on different elements.

Reply:
> parent div container

do you mean parent `li` container?

would `align-self: flex-start` be more intuitive?